walkthroughs/howto-k8s-cloudmap/deploy.sh

```diff
@@ -27,27 +27,52 @@ ECR_IMAGE_PREFIX="${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/
 FRONT_APP_IMAGE="${ECR_IMAGE_PREFIX}/feapp"
 COLOR_APP_IMAGE="${ECR_IMAGE_PREFIX}/colorapp"
 
+MANIFEST_VERSION="${1:-v1beta2}"
+
 error() {
     echo $1
     exit 1
 }
 
-check_appmesh_k8s() {
+check_virtualnode_v1beta1(){
     #check CRD
     crd=$(kubectl get crd virtualnodes.appmesh.k8s.aws -o json | jq -r '.. | .cloudMap?.properties.namespaceName? | select(. != null)')
     if [ -z "$crd" ]; then
         error "$PROJECT_NAME requires virtualnodes.appmesh.k8s.aws CRD to support Cloud Map service-discovery. See https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/CHANGELOG.md#v012"
     else
         echo "CRD check passed!"
     fi
+}
+
+check_virtualnode_v1beta2(){
+    #check CRD
+    crd=$(kubectl get crd virtualnodes.appmesh.k8s.aws -o json | jq -r '.. | .awsCloudMap?.properties.namespaceName? | select(. != null)')
+    if [ -z "$crd" ]; then
+        error "$PROJECT_NAME requires virtualnodes.appmesh.k8s.aws CRD to support Cloud Map service-discovery. See https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/CHANGELOG.md"
+    else
+        echo "CRD check passed!"
+    fi
+}
 
+check_appmesh_k8s() {
     #check aws-app-mesh-controller version
-    currentver=$(kubectl get deployment -n appmesh-system appmesh-controller -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':')
-    requiredver="v0.1.2"
+    currentver=$(kubectl get deployment -n appmesh-system appmesh-controller-manager -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':')
+
```

Comment:
We will need something similar to this patch - 
diff --git a/walkthroughs/howto-k8s-cloudmap/deploy.sh b/walkthroughs/howto-k8s-cloudmap/deploy.sh
index 55a889c..0dbe276 100755
--- a/walkthroughs/howto-k8s-cloudmap/deploy.sh
+++ b/walkthroughs/howto-k8s-cloudmap/deploy.sh
@@ -56,12 +56,13 @@ check_virtualnode_v1beta2(){
 
 check_appmesh_k8s() {
     #check aws-app-mesh-controller version
-    currentver=$(kubectl get deployment -n appmesh-system appmesh-controller-manager -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':')
 
     if [ "$MANIFEST_VERSION" = "v1beta2" ]; then
+        currentver=$(kubectl get deployment -n appmesh-system appmesh-manager -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':'|tail -n1)
         requiredver="v1.0.0"
         check_virtualnode_v1beta2
     elif [ "$MANIFEST_VERSION" = "v1beta1" ]; then
+        currentver=$(kubectl get deployment -n appmesh-system appmesh-controller -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':')
         requiredver="v0.3.0"
         check_virtualnode_v1beta1
     else

Reply:
+1, let's hold on for merging this. I'll fix it before we merge